internal/pkg/cli/env_init.go

```diff
@@ -166,6 +167,10 @@ func newInitEnvOpts(vars initEnvVars) (*initEnvOpts, error) {
 	if err != nil {
 		return nil, fmt.Errorf("read named profiles: %w", err)
 	}
+	cmd, err := newAppUpgradeOpts(appUpgradeVars{name: vars.appName})
```

Comment:
Do we have an e2e which exercises the env upgrade pathway? It might be worth it to add one so we can be sure new changes don't interfere with `env upgrade` or `app upgrade`.